components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/signup/UserSelfRegistrationManager.java

```diff
@@ -1289,11 +1289,9 @@ public boolean isUsernameAlreadyTaken(String username, String tenantDomain) thro
             tenantDomain = MultitenantUtils.getTenantDomain(username);
         }
         try {
-            String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
-
             UserRealm userRealm = getUserRealm(tenantDomain);
             if (userRealm != null) {
-                isUsernameAlreadyTaken = userRealm.getUserStoreManager().isExistingUser(tenantAwareUsername) ||
+                isUsernameAlreadyTaken = userRealm.getUserStoreManager().isExistingUser(username) ||
```

Comment:
What will happen if tenantDomain is null?

Reply:
`MultitenantUtils.getTenantDomain` function returns "carbon.super" as the default value if no specific domain could be obtained from the username. Therefore there is no scenario where the tenant domain would be null at this point.

Reply:
If tenantDomain is null, then username will tenantQualified username. but as per our discussion isExistingUser expects tenantAwareUsername.

Can't we do like this...

```
if (StringUtils.isBlank(tenantDomain)) {
            tenantDomain = MultitenantUtils.getTenantDomain(username);
            username =  MultitenantUtils.getTenantAwareUsername(username);
 }

```

Reply:
fixed with commit